main.py

```diff
@@ -178,18 +178,10 @@ async def compile_project(project_root: Path, manifest: PackageManifest):
     # Create a contracts directory
     contracts_dir = project_root / "contracts"
     contracts_dir.mkdir()
-    
-    # add request contracts in temp directory
-    if manifest.sources:
-        for filename, source in manifest.sources.items():
-            path = contracts_dir / filename
-            # NOTE: In case there is a multi-level path
-            path.parent.mkdir(parents=True, exist_ok=True)
-            path.write_text(source.fetch_content())
+    project = Project.from_manifest(manifest)
     try:
-        project = Project(project_root)
-        compiled_manifest = project.extract_manifest()
-        results[project_root.name] = compiled_manifest
+        project.load_contracts()
+        results[project_root.name] = manifest
```

Comment:
how does this line work? I feel like `manifest` isn't getting enriched

Reply:
also, can probably move it out of the `try...catch` if it's not making any calls

Reply:
So similarly to how when in local projects when you compile, the contract types get filled in the cache, the same thing here. As soon as you compile a manifest-based project, it enriches itself, because it is its own cache.

For readability, I can add `project.manifest.contract_types = project.load_contracts()` but it isn't required.

Reply:
> also, can probably move it out of the try...catch if it's not making any calls

Done!

Reply:
> So similarly to how when in local projects when you compile, the contract types get filled in the cache, the same thing here. As soon as you compile a manifest-based project, it enriches itself, because it is its own cache.
> 
> For readability, I can add `project.manifest.contract_types = project.load_contracts()` but it isn't required.

So to verify, the `manifest` is just a reference to whatever the current state of that manifest location is? It is not itself a standalone model

Reply:
I am not sure what you mean. It is more of just how the project works in Ape. Every project has a manifest. Local project's check the disk cache for their initial manifest but otherwise maintain the model in memory, updating the disk cache at certain times. A local project _is_ a manifest project, in the duck sense.

However, manifest projects are not local projects. They are just the base - a wrapper around a manifest which serves as a cache. As you compile the manifest project, it cache the contract types. You can force a re-compile using `use_cache=False`, all standard Ape stuff.

Reply:
> Local project's check the disk cache for their initial manifest but otherwise maintain the model in memory

overall it's kind of weird that we do `project.load_contracts()` to compile such that (a completely separate object) `manifest` has the compiled types, but I understand this is just how it currently is set up right now